python/ALB_HTTP_TO_HTTPS_REDIRECTION_CHECK/ALB_HTTP_TO_HTTPS_REDIRECTION_CHECK.py

```diff
@@ -0,0 +1,427 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+'''
+Rule Name:
+  ALB_HTTP_TO_HTTPS_REDIRECTION_CHECK
+
+Description:
+  Checks whether HTTP to HTTPS redirection is configured on all HTTP listeners of Application Load Balancers. The rule in NON_COMPLIANT if one or more HTTP listener of an Application Load Balancer do not have HTTP to HTTPS redirection configured.
+
+Trigger:
+  Periodic
+
+Reports on:
+  AWS::ElasticLoadBalancingV2::LoadBalancer
+
+Rule Parameters:
+  None
+
+Scenarios:
+  Scenario: 1
+     Given: No Application Load Balancer is present
+      Then: Return NOT_APPLICABLE
+  Scenario: 2
+     Given: There is at least 1 Application Load Balancer
+       And: Application Load Balancer has only HTTPS listener(s)
+      Then: Return COMPLIANT
+  Scenario: 3
+     Given: There is at least 1 Application Load Balancer
+       And: Application Load Balancer has one or more HTTP listeners configured
+       And: At least one HTTP listener rule does not have HTTP to HTTPS redirection action configured
+      Then: Return NON_COMPLIANT
+  Scenario: 4
+     Given: There is at least 1 Application Load Balancer
+       And: Application Load Balancer has one or more HTTP listeners configured
+       And: All HTTP listener rules have HTTP to HTTPS redirection action configured
+      Then: Return COMPLIANT
+'''
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::ElasticLoadBalancingV2::LoadBalancer'
+ACCOUNT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    evaluations = []
+    alb_client = get_client("elbv2", event)
+    all_elbv2 = get_all_elbv2(alb_client)
+    if not all_elbv2:
+        return [build_evaluation(event['accountId'], 'NOT_APPLICABLE', event, resource_type=ACCOUNT_RESOURCE_TYPE)]
```

Comment:
No need for a list, and no need for the global variable, since it is not used anywhere else.